cfpb/cfgov-refresh
# Df 2227 - dist directory

Moves project architecture to having a /src/ and /dist/ directory

## Additions

- Added `TEST.md` readme file for testing instructions.
- Added `grunt clean` and `grunt copy` tasks, and added `grunt clean` to `setup.sh`.

## Removals

- Removed `_tests/browser_tests/README.md`, `_tests/macro_testing/README.md`, `_tests/processor_tests/README.md`.

## Changes

- Changed project architecture to having `/src/` and `/dist/` directory.
- Changed `/_tests/` directory name to `/test/`.
- Changed `/_tests/macro_testing` directory name to `/test/macro_tests`.
- Moved `browserify-shims.js` to `/config/` directory.

## Testing

1. Run `./setup.sh`
2. Stop sheer.
3. `cd dist`
4. Start sheer.
5. ???
6. Profit. Site should function the same and tests should run.

## Review

- @dpford 
- @kurtw 
- @jimmynotjim 
- @KimberlyMunoz 
- @sebworks 
- @rosskarchner 

## Notes

- The topdoc task is picking up CSS from the /dist/ directory and generating docs for slick and pdfreactor-fonts. Since we don't use that task anyway I just put a TODO in the Gruntfile for it.
- @dpford please check that links.sh was updated correctly.
- @dpford @kurtw the processor tests are failing, but I'm not sure if they were failing before. Says, e.g.:
  ```
ERROR: test_views (tests.test_wordpress_view_processor.WordpressViewProcessorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bradforda/homebrew/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/Users/bradforda/Projects/cfgov-refresh/test/processor_tests/tests/test_wordpress_view_processor.py", line 37, in test_views
    custom_fields = document['custom_fields']
KeyError: 'custom_fields'
```
- @sebworks please check that crawler code in `/src/external-site/_vars-external-site.html` was updated correctly.
- Fixes https://github.com/cfpb/cfgov-refresh/issues/750. Full changes are in that issue.
